ts/input/tex/newcommand/NewcommandUtil.ts

```diff
@@ -320,4 +325,48 @@ export const NewcommandUtil = {
     ) as sm.EnvironmentMap;
     handler.add(env, new Macro(env, func, attr));
   },
+
+  /**
+   * Removes a user-defined macro, if there is one, and
+   * Adds an undefined macro (to block ones in later maps),
+   *   if needed.
+   *
+   * @param {TexParser} parser The current parser.
+   * @param {string} cs The control sequence to undefine.
+   */
+  undefineMacro(parser: TexParser, cs: string) {
+    const handlers = parser.configuration.handlers;
+    const macros = handlers.retrieve(NewcommandTables.NEW_COMMAND);
+    (macros as sm.CommandMap).remove(cs);
+    if (handlers.get(HandlerType.MACRO).applicable(cs)) {
+      //
+      // This will hide the macro that is in a later mapping
+      // by forcing the parser to jump directly to the fallback
+      // handler.
+      //
+      this.addMacro(parser, cs, () => SubHandler.FALLBACK, []);
+    }
+  },
+
+  /**
+   * Removes a user-defined delimiter, if there is one, and
+   * Adds an undefined one (to block ones in later maps),
+   *   if needed.
+   *
+   * @param {TexParser} parser The current parser.
+   * @param {string} cs The control sequence to undefine.
+   */
+  undefineDelimiter(parser: TexParser, cs: string) {
+    const handlers = parser.configuration.handlers;
+    const delims = handlers.retrieve(NewcommandTables.NEW_DELIMITER);
+    (delims as sm.DelimiterMap).remove(cs);
+    if (handlers.get(HandlerType.DELIMITER).applicable(cs)) {
+      //
+      // This will hide the delimiter that is in a later mapping
+      // by forcing the parser to jump directly to the fallback
+      // handler.
```

Comment:
It does not really jump to the fallback handler.

Reply:
OK, I've adjusted the comment.